### ISSUE TITLE
wayland: does not properly track keyboard and pointer focus

### ISSUE THREAD

**Issue by i509VCB** [linebender/druid#2200]
At the moment the wayland backend just tracks if a surface has focus. Using the `multiwin` example in druid, spawning another window and moving the cursor to the new window will result in the wrong window showing hover events.

Eli5: the wayland backend needs to store what surface is focused and send input events to that surface.

**Comment by james-lawrence** [comment]
it does store it it's likely just not handling the enter/leave events properly and changing the active window.
